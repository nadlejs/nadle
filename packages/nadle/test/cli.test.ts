import path from "node:path";

import { execa } from "execa";
import { it, expect, describe } from "vitest";

const cliPath = path.resolve(import.meta.dirname, "../bin/nadle");
const fixturesDir = path.resolve(import.meta.dirname, "./fixtures");

describe("CLI", () => {
	const exec = execa({ cwd: fixturesDir });

	it("prints help", async () => {
		const { stdout } = await exec`${cliPath} --help`;
		expect(stdout).toMatchSnapshot();
	});

	it("can run a simple command", async () => {
		const { stdout, exitCode } = await exec`${cliPath} hello`;
		expect(exitCode).toBe(0);
		expect(stdout).toMatchSnapshot();
	});

	it("can run two commands sequentially", async () => {
		const { stdout, exitCode } = await exec`${cliPath} hello goodbye`;
		expect(exitCode).toBe(0);
		expect(stdout).toMatchSnapshot();
	});

	it("throw error on unknown task", async () => {
		try {
			await exec`${cliPath} unknown`;
		} catch (error) {
			expect(error.exitCode).toBe(1);
			expect(error.stderr).toContain(`Error: Task "unknown" not found`);
		}
	});

	it("can run dependent task first", async () => {
		const { stdout, exitCode } = await exec`${cliPath} copy`;
		expect(exitCode).toBe(0);
		expect(stdout).toMatchSnapshot();
	});
});
