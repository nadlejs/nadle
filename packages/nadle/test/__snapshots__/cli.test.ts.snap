// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI > can run a simple command 1`] = `"Hello from nadle!"`;

exports[`CLI > can run dependent task first 1`] = `
"Preparing...
Copying from assets/ to dist/"
`;

exports[`CLI > can run two commands sequentially 1`] = `
"Hello from nadle!
Goodbye, tak!"
`;

exports[`CLI > prints help 1`] = `
"nadle [...options] <tasks...>

Run one or many tasks

Options:
      --version  Show version number                                   [boolean]
  -c, --config   Path to config file    [string] [default: <cwd>/build.nadle.ts]
      --help     Show help                                             [boolean]"
`;
